all files

tomharrigan opened the pull request:
This is for doing a complete review of the plugin
`alexa/request` and `alexa/response` and `alexaevent.php` are largely third party code, there's a fair bit of camel casing but with the camel casing from the amazon request objects, it helps with consistency in knowing what's being manipulated/used.

naming of `speechAssets` and the folders within it is for consistency among other Alexa related code in the wild

Speed is of vital importance. Wherever possible, data should be cached, and queries should be as simple as possible.

tomharrigan committed:
all files

bcampeau commented on alexa/request/application.php:
```diff
@@ -0,0 +1,40 @@
+		$this->applicationId = preg_split( '/,/', $applicationId );
+	}
+
+	public function setRequestApplicationId( $applicationId ) {
```
PHPdoc maybe

bcampeau commented on alexa/request/application.php:
```diff
@@ -0,0 +1,40 @@
+			$requestApplicationId = $this->requestApplicationId;
+		}
+		if ( ! in_array( $requestApplicationId, $this->applicationId ) ) {
+			throw new InvalidArgumentException( 'Application Id not matched' );
```
You should create a text domain and add translation functions to all of these strings.

bcampeau commented on alexa/request/certificate.php:
```diff
@@ -0,0 +1,185 @@
+
+	public $request_id;
+	public $timestamp;
+	/** @var Session */
```
Why a comment for this var and not the others?

bcampeau commented on alexa/request/certificate.php:
```diff
@@ -0,0 +1,185 @@
+	public $app_id;
+
+	/**
+	 * @param type $certificateUri
```
PHPdoc doesn't match the function

bcampeau commented on alexa/request/certificate.php:
```diff
@@ -0,0 +1,185 @@
+		$difference_in_seconds = $now->getTimestamp() - $timestamp->getTimestamp();
+
+		if ( $difference_in_seconds > self::TIMESTAMP_VALID_TOLERANCE_SECONDS ) {
+			throw new InvalidArgumentException( 'Request timestamp was too old. Possible replay attack.' );
```
I'm not going to keep commenting, but since this is a plugin I'd definitely want to see translation functions and a text domain set for the static text throughout.

bcampeau commented on alexa/request/certificate.php:
```diff
@@ -0,0 +1,185 @@
+		}
+	}
+
+	public function validate_certificate() {
```
No phpdoc and i'm therefore wondering how this function works. It doesn't seem to return anything if it's valid.

bcampeau commented on alexa/request/certificate.php:
```diff
@@ -0,0 +1,185 @@
+		$url = parse_url( $this->certificate_url );
+
+		if ( static::SIGNATURE_VALID_PROTOCOL !== $url['scheme'] ) {
+			throw new InvalidArgumentException( 'Protocol isn\'t secure. Request isn\'t from Alexa.' );
```
Use double quotes to avoid backslashing here and the lines below. Also same translation feedback.

bcampeau commented on alexa/request/certificate.php:
```diff
@@ -0,0 +1,185 @@
+	/**
+	 * Perform the actual download of the certificate
+	 */
+	public function fetch_certificate() {
```
I don't believe WP.com VIP allows curl. I had to rewrite something similar for Apple News. You probably want to see about rewriting this using the WP HTTP functions.

bcampeau commented on alexa/request/intentrequest.php:
```diff
@@ -0,0 +1,38 @@
+	public $intentName;
+	public $slots = array();
+
+	public function __construct( $raw_data, $applicationId ) {
```
phpdoc

bcampeau commented on alexa/request/request.php:
```diff
@@ -0,0 +1,99 @@
+
+	public $request_id;
+	public $timestamp;
+	/** @var Session */
```
Again weird only this one var has phpdoc

bcampeau commented on alexa/request/request.php:
```diff
@@ -0,0 +1,99 @@
+
+	/**
+	 * Set up Request with RequestId, timestamp (DateTime) and user (User obj.)
+	 * @param type $data
```
Doesn't seem to match the function

bcampeau commented on alexa/request/request.php:
```diff
@@ -0,0 +1,99 @@
+	 * Set up Request with RequestId, timestamp (DateTime) and user (User obj.)
+	 * @param type $data
+	 */
+	public function __construct( $raw_data, $applicationId = null ) {
```
Two different formats for var naming is weird. I've seen this elsewhere I think. Should be underscores generally in PHP, not camel case. Unless there's some alexa API related reason. Not a huge deal.

bcampeau commented on alexa/request/sessionendedrequest.php:
```diff
@@ -0,0 +1,13 @@
+<?php
```
Maybe add a little phpdoc here so it's clear what this is

bcampeau commented on alexa/request/user.php:
```diff
@@ -0,0 +1,13 @@
+<?php
```
Maybe add a little phpdoc here so it's clear what this is

bcampeau commented on alexa/response/reprompt.php:
```diff
@@ -0,0 +1,17 @@
+<?php
```
Maybe add a little phpdoc here so it's clear what this is

bcampeau commented on alexa/response/standardcard.php:
```diff
@@ -0,0 +1,68 @@
+		$this->content = $content;
+		if ( false === ( $images = get_transient( 'voicewp_card_image_' . $image ) ) ) {
+			$images = array(
+				'smallImageUrl' => wp_get_attachment_image_src( absint( $image ), 'alexa-small' )[0],
```
Not really code review feedback, but we added a couple thumbnail sizes for the Apple News plugins and users that still actually generate physical thumbnails and don't use photon complained about the extra volume of images. Probably not a concern for VIP but for the wider community of small sizes. Just FYI

bcampeau commented on alexa/skill/news.php:
```diff
@@ -0,0 +1,230 @@
+		if ( isset( $alexawp_settings[ $case ] ) ) {
+			$response->respond( $alexawp_settings[ $case ] );
+		} else {
+			$response->respond( __( "Sorry! I couldn't find any news about that topic. Try asking something else!", 'alexawp' ) );
```
Might want to change the text domain too when you rename the plugin

bcampeau commented on alexa/skill/news.php:
```diff
@@ -0,0 +1,230 @@
+				'title' => $single_post->post_title,
+				'image' => get_post_thumbnail_id( $id ),
+			);
+			// TODO: If content of the post changes,
```
I wouldn't call this a todo. I'd say required for launch. This could cause immediate problems on a site like People or EW if a story is republished due to a factual correction.

bcampeau commented on alexa/skill/news.php:
```diff
@@ -0,0 +1,230 @@
+				'content' => $content,
+				'ids' => $ids,
+			);
+			// TODO: hook on 'publish_*' to clear cache entry when a post type served
```
I'd call this MVP too

bcampeau commented on alexa/skill/news.php:
```diff
@@ -0,0 +1,230 @@
+			if ( ! empty( $news_posts ) && ! is_wp_error( $news_posts ) ) {
+
+				foreach ( $news_posts as $key => $news_post ) {
+					// TODO: Sounds a little strange when there's only one result.
```
Probably can wait

bcampeau commented on alexawp.php:
```diff
@@ -0,0 +1,299 @@
+ * Plugin URI: https://github.com/tomharrigan/
+ * Author: TomHarrigan
+ * Author URI: https://alexawp.com
+ * Version: 0.2
```
Since we're releasing, maybe just make it V1.0?

bcampeau commented on alexawp.php:
```diff
@@ -0,0 +1,299 @@
+
+add_action( 'init', array( 'Alexawp', 'get_instance' ), 0 );
+
+use Alexa\Request\IntentRequest;
```
I might move here below into it's own class file. This is getting a little cluttered for a main plugin file.

bcampeau commented on post-types/class-alexawp-post-type-skill.php:
```diff
@@ -0,0 +1,165 @@
+	 */
+	function update_post_metadata( $null, $post_id, $meta_key, $meta_value, $prev_value ) {
+		if ( 'alexawp_skill_type' == $meta_key ) {
+			$old_value = get_metadata( 'post', $post_id, $meta_key );
```
Why not `get_post_meta`? Just wondering

tomharrigan committed:
remove curl dependence. Use wp_remote_get

tomharrigan committed:
remove unneeded $requestApplicationId

tomharrigan committed:
remove extra line

tomharrigan committed:
add docs remove unused vars

tomharrigan committed:
we’re not using the requestID, ditch

tomharrigan committed:
phpdoc

tomharrigan committed:
rename to snake case set_certificate_dependency

and some minor phpcs fixes

tomharrigan committed:
snake case from_data

tomharrigan committed:
snake case applicationId to application_id

tomharrigan committed:
intentName to snake case intent_name

tomharrigan committed:
session_id snake case and get_attribute

tomharrigan committed:
snake case user class

tomharrigan committed:
output_speech snake case

tomharrigan committed:
should_end_session and session_attributes

tomharrigan committed:
with_card respond_ssml reprompt_ssml

tomharrigan committed:
end_session and add_session_attribute

tomharrigan committed:
add_session_attribute

tomharrigan committed:
text domain on application.php and certificate.php

tomharrigan committed:
docs

tomharrigan committed:
text domain

tomharrigan committed:
docs

tomharrigan committed:
change get_metadata to get_post_meta

tomharrigan committed:
rename to voicewp except options/meta/post types

tomharrigan committed:
updates routese to voicewp and handles legacy

tomharrigan committed:
update endpoints in readme

tomharrigan committed:
fix briefings

I broke briefings when I made the autoloader ‘better’

tomharrigan committed:
these fields sometimes don’t play well

ditching these labels because depending on FM version they throw errors and this is just too annoying to deal with

tomharrigan committed:
a back compat fix

handle cases where class doesnt exist with skill type

tomharrigan committed:
bump min wp version for unregister_post_type

tomharrigan committed:
voicewp all the things

tomharrigan committed:
bring compat back down

ditched the unregister post type function, caused a fatal due to the rewrites and how early we hook.

tomharrigan committed:
handle migrating to new version with new naming

tomharrigan committed:
move classes out to own files

tomharrigan committed:
doc updates

tomharrigan committed:
Version updates

tomharrigan committed:
fix issue with constant scope

tomharrigan committed:
cache sprucing

tomharrigan committed:
tab fixes

tomharrigan committed:
fix meta key name

tomharrigan committed:
simplify logic
